Test/Integration/Search/SearchTest.php

```diff
@@ -5,21 +5,101 @@
 use Algolia\AlgoliaSearch\Helper\Data;
 use Algolia\AlgoliaSearch\Model\Indexer\Product;
 use Algolia\AlgoliaSearch\Test\Integration\TestCase;
+use Magento\Framework\Exception\NoSuchEntityException;
 
 class SearchTest extends TestCase
 {
-    public function testSearch()
+    /** @var Product */
+    protected $productIndexer;
+
+    /** @var Data */
+    protected $helper;
+
+    public function setUp(): void
     {
-        /** @var Product $indexer */
-        $indexer = $this->getObjectManager()->create(Product::class);
-        $indexer->executeFull();
+        parent::setUp();
 
+        $this->productIndexer = $this->objectManager->get(Product::class);
+        $this->helper = $this->getObjectManager()->create(Data::class);
+
+        $this->productIndexer->executeFull();
         $this->algoliaHelper->waitLastTask();
+    }
 
-        /** @var Data $helper */
-        $helper = $this->getObjectManager()->create(Data::class);
-        list($results, $totalHits, $facetsFromAlgolia) = $helper->getSearchResult('', 1);
+    public function testSearch()
+    {
+        $query = 'bag';
+        $results = $this->search($query);
+        $result = $this->getFirstResult($results);
+        // Search returns result
+        $this->assertNotEmpty($result, "Query didn't bring result");
+
+        $product = $this->objectManager->create(\Magento\Catalog\Model\Product::class);
+        $product->load($result['entity_id']);
+        // Result exists in DB
+        $this->assertNotEmpty($product->getName(), "Query result item couldn't find in the DB");
+        // Query word exists title
+        $this->assertStringContainsString($query, strtolower($product->getName()), "Query word doesn't exist in product name");
+    }
+
+    public function testSearchBySku()
+    {
+        $sku = "24-MB01";
+        $results = $this->search($sku);
+        $result = $this->getFirstResult($results);
+        // Search by SKU returns result
+        $this->assertNotEmpty($result, "SKU search didn't bring result");
 
-        $this->assertNotEmpty($results);
+        $product = $this->objectManager->create(\Magento\Catalog\Model\Product::class);
+        $product->load($result['entity_id']);
+        // Result exists in DB
+        $this->assertNotEmpty($product->getSku(), "SKU search result item couldn't find in the DB");
+        // Query word exists title
+        $this->assertEquals($sku, $product->getSku(), "Query SKU doesn't match with product SKU");
+    }
+
+    public function testCategorySearch()
+    {
+        // Get products by categoryId
+        list($results, $totalHits, $facetsFromAlgolia) = $this->search('', 1, [
+            'facetFilters' => ['categoryIds:' . $this->assertValues->expectedCategory]
```

Comment:
`expectedCategory` is in fact the expected number of categories indexed in the Algolia dashboard and not an actual ID . I guess `$this->assertValues->expectedCategory` is quite missleading 😄 . 

Feel free to use a `const` in the class to arbitrary search for a category id (like we did in newly created tests) : https://github.com/algolia/algoliasearch-magento-2/blob/release/3.14.3/Test/Integration/Category/MultiStoreCategoriesTest.php#L31